configure.ac

```diff
@@ -21,6 +21,7 @@ AC_LANG([C])
 
 AC_CONFIG_HEADER([config.h])
 AC_PROG_CC
+AC_PROG_CC_STDC
```

Comment:
Yeah, without that line, a lot of your code fails.  From the commit message (though I should probably have added a comment):

> Updated configure.ac to have the AC_PROG_CC_STDC macro, which uses std99 and stops errors like, " ‘for’ loop initial declarations are only allowed in C99 mode"